### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by kobizz**

### CODE REVIEW

File: docs/content/controls/README.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+Default settings in the base class:
+
+```php
+abstract class Control_Base {
+
+  // here is some base default settings
+  private $_base_settings = [
+     'separator' => 'default',
+     'label' => ‘’,
```
Comment (iamntz) [inline docs/content/controls/README.md]: Maybe it's a good idea to search for `‘` and `’` (apostrophes?) and replace them with `'` (single quote)?

Comment (KingYes) [inline docs/content/controls/README.md]: You are right.

@cobicarmel  please fix this

File: docs/content/controls/_checkbox.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+*Returns* `string|empty` Returns the string `on` if checked or empty if unchecked
+
+##Example
+
+```php
+$this->add_control(
+	'show_title',
+	[
+		'label' => __( 'Show Title', 'your-plugin' ),
+		'type' => Controls_Manager::SWITCHER,
```
Comment (iamntz) [inline docs/content/controls/_checkbox.md]: Since this is a checkbox, maybe this should read `Controls_Manager::CHECKBOX` instead?

File: docs/content/controls/_select2.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  'show_elements',
+  [
+     'label'       => __( 'Show Elements', 'your-plugin' ),
+     'type' => Controls_Manager::SELECT2,
+     'options' => [
+     	'title'       => __( 'Title', 'your-plugin' ),
+     	'desctiption' => __( 'Desctiption', 'your-plugin' ),
+     	'button'      => __( 'Button', 'your-plugin' ),
+     ],
+	 'multiple' => 'true',
```
Comment (iamntz) [inline docs/content/controls/_select2.md]: I think `'multiple' => true` should be better (use `true` as a boolean, not as a string)

File: docs/content/controls/_gallery.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+#Gallery
+A control that allows the user to choose set of images from the WordPress media library. 
+
+*Returns:* An array of single image arrays:
+```php
+array(
```
Comment (iamntz) [inline docs/content/controls/_gallery.md]: The whole plugin uses the square brance format for arrays; I think docs should reflet the same code style.

File: docs/content/controls/_select2.md (hunk 3 above)
Comment (KingYes) [inline docs/content/controls/_select2.md]: Nope. It's just string to move JS codes

File: docs/content/controls/_gallery.md (hunk 4 above)
Comment (KingYes) [inline docs/content/controls/_gallery.md]: It's for readable the return value..

File: docs/content/controls/_select2.md (hunk 3 above)
Comment (iamntz) [inline docs/content/controls/_select2.md]: right, but what `'multiple' => 'false'` would do in javascript? 

In this context, to be more specific:

```
<# var multiple = ( data.multiple ) ? 'multiple' : ''; #>
```

Comment (KingYes) [inline docs/content/controls/_select2.md]: Good point. But the condition is for `'multiple' => ''` (default value).

File: docs/content/controls/_select2.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+```php
+$this->add_control(
+  'show_elements',
+  [
+     'label'       => __( 'Show Elements', 'your-plugin' ),
+     'type' => Controls_Manager::SELECT2,
+     'options' => [
+     	'title'       => __( 'Title', 'your-plugin' ),
+     	'desctiption' => __( 'Desctiption', 'your-plugin' ),
```
Comment (iamntz) [inline docs/content/controls/_select2.md]: `desctiption` vs `description` ? :)

File: docs/content/controls/_gallery.md [hunk 6]
```diff
@@ -26,10 +26,10 @@ $this->add_control(
 
 **PHP** *(Under `render()` method)*
 ```php
-$images = $this->get_settings( ‘gallery’ );
+$images = $this->get_settings( 'gallery' );
 
 foreach ( $images as $image ) {
-    echo ‘<img src=”’ . $image[url] . ‘“>’;
+    echo '<img src=”' . $image[url] . '“>';
```
Comment (iamntz) [inline docs/content/controls/_gallery.md]: You forgot the doubles: `”` 💃

Comment (cobicarmel) [inline docs/content/controls/_gallery.md]: Thanks

File: includes/base/element-base.php [hunk 7]
```diff
@@ -116,7 +116,7 @@ public function get_controls( $control_id = null ) {
 		return self::_get_items( $stack['controls'], $control_id );
 	}
 
-	public function add_control( $id, $args ) {
+	public function add_control( $id, array $args ) {
```
Comment (iamntz) [inline includes/base/element-base.php]: This type of commit shouldn't be on a different PR though? It has nothing to do with the docs..

Comment (joshmarom) [inline includes/base/element-base.php]: Self documenting code ;)

Comment (iamntz) [inline includes/base/element-base.php]: Hardly.

Adding type hinting could break existing code if it's added late in the project.

Not saying it's a bad thing to have, but surely isn't a thing that should appear in _docs_ commits :)
